x-pack/plugins/observability_solution/infra/public/pages/metrics/inventory_view/hooks/use_snaphot.ts

```diff
@@ -19,79 +16,74 @@ import {
 
 export interface UseSnapshotRequest
   extends Omit<SnapshotRequest, 'filterQuery' | 'timerange' | 'includeTimeseries'> {
-  filterQuery: string | null | symbol | undefined;
+  filterQuery?: string | null | symbol;
   currentTime: number;
-  sendRequestImmediately?: boolean;
   includeTimeseries?: boolean;
   timerange?: InfraTimerangeInput;
-  requestTs?: number;
 }
 
-export interface UseSnapshotRequestOptions {
-  abortable?: boolean;
+export function useSnapshot(
+  props: UseSnapshotRequest,
+  { sendRequestImmediately = true }: { sendRequestImmediately?: boolean } = {}
+) {
+  const payload = useMemo(() => JSON.stringify(buildPayload(props)), [props]);
+
+  const { data, status, error, refetch } = useFetcher(
+    async (callApi) => {
+      const response = await callApi('/api/metrics/snapshot', {
+        method: 'POST',
+        body: payload,
+      });
+
+      return decodeOrThrow(SnapshotNodeResponseRT)(response);
+    },
+    [payload],
+    {
+      autoFetch: sendRequestImmediately,
+    }
+  );
+
+  return {
+    error: (error && error.message) || null,
+    loading: isPending(status),
+    nodes: data ? data.nodes : [],
+    interval: data ? data.interval : '60s',
```

Comment:
Can't you do it like this?
```suggestion
    nodes: data?.nodes || [],
    interval: data?.interval || '60s',
```